index-handlers/src/main/java/no/sikt/nva/nvi/index/utils/NviCandidateIndexDocumentGenerator.java

```diff
@@ -0,0 +1,144 @@
+    }
+
+    private static Affiliation createAffiliation(JsonNode affiliation) {
+        return new Affiliation(affiliation.get(FIELD_ID).textValue(),
+                               convertToMap(affiliation.get(FIELD_LABELS)),
+                               ApprovalStatus.PENDING.getValue());
+    }
+
+    private static Map<String, String> convertToMap(JsonNode node) {
+        return dtoObjectMapper.convertValue(node, new TypeReference<>() {
+        });
+    }
+
+    private static String extractPublicationId(JsonNode resource) {
+        return resource.at(JSON_PTR_ID).textValue();
+    }
+
+    private static Stream<JsonNode> getJsonNodeStream(JsonNode jsonNode, String jsonPtr) {
+        return StreamSupport.stream(jsonNode.at(jsonPtr).spliterator(), false);
+    }
+
+    private static String extractPublicationIdentifier(JsonNode resource) {
+        return UriWrapper.fromUri(extractPublicationId(resource)).getPath().getLastPathElement();
+    }
+
+    private static PublicationDetails extractPublication(JsonNode resource) {
+        return new PublicationDetails(extractPublicationId(resource), extractInstanceType(resource),
+                                      extractMainTitle(resource), extractPublicationDate(resource),
+                                      extractContributors(resource));
+    }
+
+    private static List<Contributor> extractContributors(JsonNode resource) {
+        return getJsonNodeStream(resource, JSON_PTR_CONTRIBUTOR)
+                   .map(contributor -> contributor.get(FIELD_IDENTITY))
+                   .map(NviCandidateIndexDocumentGenerator::createContributor)
+                   .toList();
+    }
+
+    private static Contributor createContributor(JsonNode identity) {
+        return new Contributor(identity.get(FIELD_ID).textValue(),
+                               identity.get(FIELD_NAME).textValue(),
+                               identity.get(FIELD_ORCID).textValue());
+    }
+
+    private static String extractPublicationDate(JsonNode resource) {
+        var publicationDate = resource.at(JSON_PTR_PUBLICATION_DATE);
+        return formatPublicationDate(publicationDate);
+    }
+
+    private static String extractMainTitle(JsonNode resource) {
+        return resource.at(JSON_PTR_MAIN_TITLE).textValue();
+    }
+
+    private static String extractInstanceType(JsonNode resource) {
+        return resource.at(JSON_PTR_INSTANCE_TYPE).textValue();
+    }
+
+    private static String extractYear(JsonNode resource) {
+        return resource.at(JSON_PTR_PUBLICATION_DATE_YEAR).textValue();
+    }
+
+    private static String formatPublicationDate(JsonNode publicationDateNode) {
+        var year = publicationDateNode.get(FIELD_PUBLICATION_DATE_YEAR);
+        var month = publicationDateNode.get(FIELD_PUBLICATION_DATE_MONTH);
+        var day = publicationDateNode.get(FIELD_PUBLICATION_DATE_DAY);
+
+        return attempt(() -> LocalDate.of(year.asInt(), month.asInt(), day.asInt()).toString()).orElse(
```

Comment:
The safest way of doing this — I think — is treating it as an ISO 8601 string with simple concatenation c.f. https://www.w3.org/TR/NOTE-datetime#:~:text=options%20it%20permits.-,Formats,-Different%20standards%20may

Hence: 2000, 2000-01, 2000-01-01

The trouble is that these are strings and that's all — it's deliberate. We observe a some weird stuff in these fields on import. We also assume Gregorian dates which will bite us at some point.